### DISCUSSION TITLE
PowerLimiter: Verständnis Problem

### DISCUSSION THREAD

**Discussion by Adminius** [hoylabs/OpenDTU-OnBattery#169] [Q&A] [answered]
Hardware Testaufbau:

- HM350, (HM600 wird dann im Produktiveinsatz sein).
- Akku 15S 20Ah LiFePo4
- Victron 100/20 48V
- Es sind keine PV-Module angeschlossen => immer Nacht
- PowerMeter über MQTT alle 3 Phasen einzeln

Einstellungen:
![image](https://user-images.githubusercontent.com/17052262/231379837-81804099-3763-4ab9-b477-91ac5c07f160.png)
![image](https://user-images.githubusercontent.com/17052262/231379961-b8d3d33c-95d1-4cf8-8c7d-2f4822723a4e.png)

Akkuspannung ist über 49V.

heute Nacht:
![image](https://user-images.githubusercontent.com/17052262/231383481-cf72a64e-0124-45d3-89d3-d390da925b55.png)
Warum nicht die ganze Zeit die ca. 100W Einspeisen?
Das sagte der Log dazu:
![image](https://user-images.githubusercontent.com/17052262/231383849-053240eb-cb44-4b39-af6c-7c16515b38df.png)

Und "nicht einspeisen" ging so die ganze Nacht, bis wir aufgewacht sind. Da frage ich mich warum er 365W erzeugen will obwohl ich 200W gesetzt habe oder gilt da die 800W Grenze, oder liege ich total daneben? Was macht dann die 200W grenze und warum speist er bei 100W nichts ein?
![image](https://user-images.githubusercontent.com/17052262/231384868-e4eb75e3-b716-4d3a-abd9-a78b1c141142.png)

**Comment by swingstate** [comment]
Vielleicht helfen die Settings [hier](https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/152#discussioncomment-5543039) weiter?

**Comment by berni2288** [comment, marked as answer]
Hi @Adminius,

die Einstellung `Erlaubter Stromverbrauch` ist leider etwas unglücklich benannt und nicht gut erklärt. Das gehört meiner Meinung nach überarbeitet. Besser wäre eine Bezeichnung wie `Permanenter Netzbezug` oder so.
Der Wert gibt nämlich an wie viel Strom immer aus dem Netz bezogen werden soll. Ich glaube das gibt es deshalb, damit bei kurzzeitigen Schwankungen im Stromverbrauch nicht unnötig zuviel Energie aus der Batterie geholt wird.
Ich glaube nicht dass du das auf `200` setzen willst. Ich habe es bei mir auf `10` gesetzt.
Deswegen war das Limit bei dir immer um 200 zu niedrig.

**Reply by Adminius** [reply]
ok, das macht Sinn. "EN: Target power consumption" macht mehr Sinn, da müsste man auch "grid" ergänzen.
ich vermisse immer noch die Einstellung: wenn 0 PV dann max. XY Watt aus dem Akku, sonst bis "Oberes Limit". Man sieht, dass die Akkus-Spannung recht stark einbricht wenn man viel Leistung zieht. Da würde ich auf 50-100 Watt aus dem Akku beschränken, tagsüber darf aber die volle Leistung (z.B. 600W beim HM600) gehen.

**Reply by Adminius** [reply]
ok, jetzt sieht es besser aus:
![image](https://user-images.githubusercontent.com/17052262/231415123-8c4ea8d3-b98e-4823-87a8-33f19e2d7595.png)
ich habe "Erlaubter Stromverbrauch:" auf 5 Watt gesetzt. 

Ich würde vorschlagen:
Erlaubter Netzbezug
Target grid consumption

**Reply by berni2288** [reply]
`Erlaubter` ist für mich nicht erklärend genug. "Erlaubt" kann viel bedeuten. Außerdem sollte die Englischübersetzung ca. dasselbe bedeuten. In der Beschreibung sollte auch erklärt sein warum man das überhaupt braucht und warum das größer als 0 sein sollte.

Spannungseinbruch: Das ist normal unter Last, dafür gibts den Korrekturfaktor, der normalisierte Wert wird aber glaub ich im MQTT und in der API nicht angezeigt und nur intern verwendet.

**Reply by Adminius** [reply]
Spannungseinbruch ist in den Grenzregionen (unter 3.0V/Zelle) ein Problem. 
Ich sehe ehrlich gesagt auch wenig Sinn den Akku möglichst schnell mit max. Leistung=max.Leitungsverlusten zu entladen. 
Da wäre mir eine "konstante" Leistung (Also bis XY Watt) lieber. Dann kann man bei kleinem SoC mehr aus dem Akku holen, da die Spannung bei kleineren Strömen nicht so stark einbricht und somit länger genutzt werden kann.

Nächster Vorschlag:
Ziel/Tolerierter Netzbezug: Tolerierte minimale Leistung die aus dem Netz bezogen werden darf.

**Reply by berni2288** [reply]
Ja, also ich denke jeder hat hier andere use cases und Setups, meiner Meinung nach sollten wir hier einfach möglichst viele Konfigurationsoptionen anbieten, die müssen aber alle gut verständlich und gut beschrieben sein.

Danke für die Vorschläge. Ich wollt die Labels und Beschreibungen eh mal alle überarbeiten :)
